shared/src/main/scala/LiterateLexer.scala

```diff
@@ -0,0 +1,158 @@
+    """\(""".r ~ rep(normalGroup | """[^()]+""".r) ~ """\)""".r ^^ {
+      case _ ~ body ~ _ => Word(sbcsify(body.map(recHelp).mkString))
+    }
+
+  def list: Parser[LiterateToken] =
+    "[" ~ repsep(list | """[^\]\[|]+""".r, "|") ~ "]" ^^ { case _ ~ body ~ _ =>
+      ListToken(body)
+    }
+
+  def word: Parser[LiterateToken] =
+    """[a-zA-Z?!*+=&%><-][a-zA-Z0-9?!*+=&%><-]*""".r ^^ { value =>
+      Word(value)
+    }
+
+  def varGet: Parser[LiterateToken] = """\$([_a-zA-Z][_a-zA-Z0-9]*)?""".r ^^ {
+    value => AlreadyCode("#" + value)
+  }
+
+  def varSet: Parser[LiterateToken] = """:=([_a-zA-Z][_a-zA-Z0-9]*)?""".r ^^ {
+    value => AlreadyCode("#" + value.substring(1))
+  }
+
+  def augVar: Parser[LiterateToken] = """:>([a-zA-Z][_a-zA-Z0-9]*)?""".r ^^ {
+    value => AlreadyCode("#>" + value.substring(2))
+  }
+
+  def unpackVar: Parser[LiterateToken] = ":=" ~ list ^^ { case _ ~ value =>
+    (value: @unchecked) match
+      case ListToken(value) =>
+        AlreadyCode("#:" + value.map(recHelp).mkString("[", "|", "]"))
+  }
+
+  def branch: Parser[LiterateToken] = "|" ^^ { value =>
+    AlreadyCode("|")
+  }
+
+  def rawCode: Parser[LiterateToken] = "#([^#]|#[^}])*#}".r ^^ { value =>
+    AlreadyCode(value.substring(1, value.length - 2))
+  }
+
+  def tokens: Parser[List[LiterateToken]] = phrase(
+    rep(
+      number | string | singleCharString | comment | rawCode | list | lambdaBlock | normalGroup | unpackVar | varGet | varSet | augVar | word | branch
+    )
+  )
+
+  def apply(code: String): Either[VyxalCompilationError, List[LiterateToken]] =
+    (parse(tokens, code): @unchecked) match
+      case NoSuccess(msg, next)  => Left(VyxalCompilationError(msg))
+      case Success(result, next) => Right(result)
+end LiterateLexer
+
+def recHelp(token: Object): String =
+  token match
+    case Word(value)        => value
+    case AlreadyCode(value) => value
+    case LitComment(value)  => value
+    case LambdaBlock(value) => value.map(recHelp).mkString("λ", " ", "}")
+    case ListToken(value)   => value.map(recHelp).mkString("[", "|", "]")
+    case value: String      => value
+
+def sbcsify(tokens: List[LiterateToken]): String =
+  tokens.map(sbcsify).mkString("", " ", "")
+
+def sbcsify(token: Object): String =
+  token match
+    case Word(value) =>
+      literateModeMappings.getOrElse(
+        value,
+        hardcodedKeywords.getOrElse(value, value)
+      )
+    case AlreadyCode(value) => value
+    case LitComment(value)  => ""
+    case LambdaBlock(value) => value.map(sbcsify).mkString("λ", " ", "}")
+    case ListToken(value)   => value.map(sbcsify).mkString("#[", "|", "#]")
+    case value: String      => litLex(value)
+
+def getRight(
+    either: Either[VyxalCompilationError, List[LiterateToken]]
+): List[LiterateToken] =
+  // For when you know it's a right and don't want to have 20 thousand match statements
+  either match
+    case Right(value) => value
+    case Left(value) =>
+      List()
```

Comment:
```suggestion
  either.getOrElse(Nil)
```

Reply:
Perhaps emanresu A was right in that discussion we had a while back about Eithers vs exceptions. If using Eithers is being very annoying, we could throw `VyxalCompilationError`s instead.